src/server/auth/server/api_server.go

```diff
@@ -1694,25 +1743,30 @@ func (a *apiServer) GetAuthToken(ctx context.Context, req *auth.GetAuthTokenRequ
 		if !isAdmin && req.Subject != callerInfo.Subject {
 			return nil, &auth.ErrNotAuthorized{
 				Subject: callerInfo.Subject,
-				AdminOp: "GetOneTimePassword on behalf of another user",
+				AdminOp: "GetAuthToken on behalf of another user",
 			}
 		}
 	}
 
 	// Compute TTL for new token that the user will get once OTP is exchanged
 	// Note: Admins always use default TTL (30 days currently), which means they
 	// can extend their session by getting an OTP and exchanging it for a new
-	// token with a later expiration than their curren token
-	var ttl = int64(defaultTokenTTLSecs)
+	// token with a later expiration than their curren token. See duplicate code
+	// in GetOneTimePassword for an explanation
 	if !isAdmin {
 		// Caller is getting OTP for themselves--use TTL of their current token
-		ttl, err = a.getCallerTTL(ctx)
+		ttl, err := a.getCallerTTL(ctx)
 		if err != nil {
 			return nil, err
 		}
-		if req.TTL == 0 || req.TTL > ttl {
+		if req.TTL <= 0 || req.TTL > ttl {
 			req.TTL = ttl
 		}
+	} else if version.IsAtLeast(1, 10) && req.TTL == 0 || req.TTL > defaultSessionTTLSecs {
```

Comment:
I'm not sure what the precedence rules are here between `&&` and `||` - could you clarify with parens?

Reply:
Done